vsg/__main__.py

```diff
@@ -393,13 +409,21 @@ def main():
 
     dStyle = read_predefined_style(commandLineArguments.style)
 
-    configuration = read_configuration_files(dStyle, commandLineArguments)
+    fix_only = settings.open_configuration_file(commandLineArguments.fix_only)
+
+    if commandLineArguments.configuration:
+        configuration = read_configuration_files(dStyle, commandLineArguments)
+
+        dIndent = read_indent_configuration(configuration)
```

Comment:
`read_indent_configuration` modifies the configuration and also returns a part of that configuration. Are both necessary? (IMO I think a function should either return a value, or modify one of the arguments but not both. I did not expect `read_indent_configuration` would modify its argument)

Reply:
hmmm...so `read_indent_configuration` essentially modifies the existing configuration and then returns a subset of it.  The subset is then used on this line:
```python
oVhdlFile.set_indent_map(dIndent)
```
Which could be changed to:
```python
oVhdlFile.set_indent_map(configuration['indent'])
```
Then you could remove the return from `read_indent_configuration`